Framework/Muon/test/MuonPreProcessTest.h

```diff
@@ -0,0 +1,435 @@
+
+  void test_rebinning_with_fixed_bin_widths_produces_correct_x_and_y_values() {
+    // x =  0.0 , 0.1 , 0.2 , ... , 1.0 (bin edges)
+    // y =  0   , 1   , 2   , ... , 9
+    auto ws = createCountsWorkspace(2, 10, 0.0);
+    setUpADSWithWorkspace setup(ws);
+    IAlgorithm_sptr alg =
+        algorithmWithoutOptionalPropertiesSet(setup.inputWSName);
+    std::vector<double> rebinArgs = {0.2};
+    alg->setProperty("RebinArgs", rebinArgs);
+    alg->setProperty("OutputWorkspace", "__notUsed");
+    alg->setAlwaysStoreInADS(false);
+    alg->execute();
+
+    WorkspaceGroup_sptr outputWS;
+    outputWS = alg->getProperty("OutputWorkspace");
+
+    MatrixWorkspace_sptr wsOut =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(outputWS->getItem(0));
+
+    TS_ASSERT_DELTA(wsOut->readX(0)[0], 0.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[1], 0.200, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[4], 0.800, 0.001);
+
+    TS_ASSERT_DELTA(wsOut->readY(0)[0], 1.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[1], 5.00, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[4], 17.000, 0.001);
+  }
+
+  void
+  test_rebinning_with_logarithmic_binning_produces_correct_x_and_y_values() {
+    // x =  1.0 , 1.1 , 1.2 , ... , 2.0 (bin edges)
+    // y =  0   , 1   , 2   , ... , 9
+    auto ws = createCountsWorkspace(1, 10, 0.0, 0, true, 1.0, 2.0);
+    setUpADSWithWorkspace setup(ws);
+    IAlgorithm_sptr alg =
+        algorithmWithoutOptionalPropertiesSet(setup.inputWSName);
+    std::vector<double> rebinArgs = {-0.2};
+    alg->setProperty("RebinArgs", rebinArgs);
+    alg->setProperty("OutputWorkspace", "__notUsed");
+    alg->setAlwaysStoreInADS(false);
+    alg->execute();
+
+    WorkspaceGroup_sptr outputWS;
+    outputWS = alg->getProperty("OutputWorkspace");
+
+    MatrixWorkspace_sptr wsOut =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(outputWS->getItem(0));
+
+    // Using "FullBinsOnly" as false in Rebin preserves
+    // the counts at the expense of an uneven bin
+    // at the end of the range, as seen below.
+    TS_ASSERT_DELTA(wsOut->readX(0)[0], 1.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[1], 1.200, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[2], 1.440, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[3], 1.728, 0.001);
+    TS_ASSERT_DELTA(wsOut->readX(0)[4], 2.000, 0.001);
+
+    TS_ASSERT_DELTA(wsOut->readY(0)[0], 1.000, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[1], 6.600, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[2], 15.360, 0.001);
+    TS_ASSERT_DELTA(wsOut->readY(0)[3], 22.040, 0.001);
+  }
+
+  // --------------------------------------------------------------------------
+  // Correct output : Time offset
+  // --------------------------------------------------------------------------
+
+  void test_that_positive_time_offset_applied_correctly() {
+    // x =  0.0 , 0.1 , 0.2 , ... , 1.0 (bin edges)
+    // y =  0   , 1   , 2   , ... , 9
+    auto ws = createCountsWorkspace(1, 10, 0.0);
+
```

Comment:
could we have an offset parameter and the repeat it in the below line and in the checks?